components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -20,23 +20,36 @@ namespace scream
 // according to the "averaging" type, and according to the number of
 // model time steps since the last output step.
 KOKKOS_INLINE_FUNCTION
-void combine (const Real& new_val, Real& curr_val, const OutputAvgType avg_type)
+void combine (const Real& new_val, Real& curr_val, Real& avg_coeff, const OutputAvgType avg_type, const Real fill_value)
 {
-  switch (avg_type) {
-    case OutputAvgType::Instant:
-      curr_val = new_val;
-      break;
-    case OutputAvgType::Max:
-      curr_val = ekat::impl::max(curr_val,new_val);
-      break;
-    case OutputAvgType::Min:
-      curr_val = ekat::impl::min(curr_val,new_val);
-      break;
-    case OutputAvgType::Average:
-      curr_val += new_val;
-      break;
-    default:
-      EKAT_KERNEL_ERROR_MSG ("Unexpected value for m_avg_type. Please, contact developers.\n");
+  const bool new_fill  = new_val == fill_value;
+  const bool curr_fill = curr_val == fill_value;
+  if (!new_fill) {
+    avg_coeff += 1;
+  }
+  if (curr_fill & new_fill) {
```

Comment:
Did you mean `&&`?

Reply:
Good catch

Reply:
FYI, both `&` and `&&` work correctly for bool types in C++ (not necessarily in C!). However, it's true that someone may be puzzled by this, and maybe think that some nifty bitwise check is needed here, so the `&&` operator is probably cleaner.